### PR TITLE
[SWTCH-299] Add filter enrollment requests

### PR DESCRIPTION
**Pull request by DawidGil-jlabs**
This PR is created by the request of Vadym.

### CODE REVIEW

**Review by jrhender — CHANGES_REQUESTED** [review]
@VOSpdUkraine I am getting an exception when navigating to the enrolment page for assets when running this branch. The exception doesn't seem to occur when trying the same action on https://switchboard-dev.energyweb.org/ Could you please confirm that this filtering works for assets as well when you have a moment?
image
```
ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'value' of undefined
TypeError: Cannot read property 'value' of undefined
    at EnrolmentListComponent.<anonymous> (enrolment-list.component.ts:143)
    at Generator.next (<anonymous>)
```

**Comment by VOSpdUkraine** [comment]
Hi @jrhender, Yes, thanks, I missed this exception, I fixed it

**Review by jrhender — APPROVED** [review]
Looks good to me

File: src/app/routes/enrolment/enrolment-list/enrolment-list.component.ts [hunk 1]
```diff
@@ -268,4 +274,27 @@ export class EnrolmentListComponent implements OnInit, OnDestroy {
...
+    if (!this.namespaceFilterControl) {
+      return;
+    }
+
+    this.namespaceFilterControl.valueChanges
+        .pipe(
+            distinctUntilChanged((prevValue, currentValue) => prevValue === currentValue),
+            takeUntil(this._subscription$)
+        )
+        .subscribe((filterValue) => {
```
Comment (DawidGil-jlabs) [inline src/app/routes/enrolment/enrolment-list/enrolment-list.component.ts:288]: you don't need here `{ ...}`. It's fine to write `(filterValue) => this_updateList(filterValue)`

**Pull request merged**